### ISSUE TITLE
issue accessing `provider` or `receiver` on an EconomicEvent

### ISSUE THREAD

**Issue by Connoropolous** [holo-rea/holo-rea#199]
RESULTS of tracing... 

The issue is clear, `id`s are being passed to `agent` queries passing not just the AgentPubKey, but the dna address too, but  the `agent-registration` zome does not expect this. 

This looks like it could be solved in one centralized place:
https://github.com/holo-rea/holo-rea/blob/1ef46ddd53380338f69383b0094025fdb855e6f4/modules/vf-graphql-holochain/queries/agent.ts#L51-L54

I see two options
1. A short term fix could just be commenting out the zome call, since it's not currently adding value anyways.
2. Perform a zomeCall where the special `encodeFields` step is not performed. Then, just split and deserialize

```javascript
      const agentId = deserializeHash(id.split(':')[0])
      const isAgent = await agentExistsNoEncoding({ pubKey: agentId })
```

The target serialization is: https://github.com/holochain-open-dev/agent-registration/blob/1843392ef5fca06138530631ad5de8c90800a64b/zome/rpc/src/lib.rs#L8

p.s. this is a useful tool: https://kawanet.github.io/msgpack-lite/

log says
```
[2022-02-24T18:43:51.807239-08:00] INFO - [HOLOCHAIN] [2mFeb 24 18:43:51.803[0m [31mERROR[0m [1mwasm_trace[0m: hc_zome_agent_registration::__is_registered_extern:/home/runner/work/holo-rea/holo-rea/.cargo/git/checkouts/agent-registration-e4baaad88c795da6/a37cc26/zome/zome/src/lib.rs:24 output_type = "core::result::Result<bool, holochain_wasmer_common::result::WasmError>"; bytes = [129, 166, 112, 117, 98, 75, 101, 121, 146, 196, 39, 132, 45, 36, 235, 142, 179, 94, 68, 10, 54, 240, 180, 190, 17, 36, 67, 21, 190, 136, 104, 40, 32, 177, 184, 249, 11, 107, 117, 76, 131, 53, 220, 232, 9, 180, 100, 56, 197, 206, 196, 39, 132, 32, 36, 50, 184, 80, 188, 156, 231, 1, 22, 212, 232, 113, 167, 14, 183, 76, 147, 231, 140, 108, 64, 196, 236, 31, 6, 156, 16, 192, 235, 172, 2, 191, 68, 120, 81, 126, 159]; Deserialize("invalid type: byte array, expected u8")
```

<img width="1680" alt="Screen Shot 2022-02-24 at 6 45 18 PM" src="https://user-images.githubusercontent.com/1409121/155644140-086d63c0-dc4f-4daf-829f-2b6830dac398.png">

I am taking a look into it

**Comment by Connoropolous** [comment]
tracing the issue, I see its sending the whole 'identifier' which includes the dnahash

```
{
  "pubKey": [
    {
      "type": "Buffer",
      "data": [
        132,
        45,
        36,
        235,
        142,
        179,
        94,
        68,
        10,
        54,
        240,
        180,
        190,
        17,
        36,
        67,
        21,
        190,
        136,
        104,
        40,
        32,
        177,
        184,
        249,
        11,
        107,
        117,
        76,
        131,
        53,
        220,
        232,
        9,
        180,
        100,
        56,
        197,
        206
      ]
    },
    {
      "type": "Buffer",
      "data": [
        132,
        32,
        36,
        50,
        184,
        80,
        188,
        156,
        231,
        1,
        22,
        212,
        232,
        113,
        167,
        14,
        183,
        76,
        147,
        231,
        140,
        108,
        64,
        196,
        236,
        31,
        6,
        156,
        16,
        192,
        235,
        172,
        2,
        191,
        68,
        120,
        81,
        126,
        159
      ]
    }
  ]
}
```
but the system is just expecting an `AgentPubKey`

**Comment by pospi** [comment]
Ah yeah, this is where my pursuit of mixing REA-specific identifiers and idiomatic Holochain IDs for zomes remixable elsewhere is getting fiddly. I consider everything related to `Agent` temporary anyway, until we've decided on an approach for #172.

I tried a couple of things:

```ts
    agent: injectTypename('Person', async (root, { id }): Promise<Agent> => {
      const pubKey = Buffer.from(deserializeHash(id.split(':')[0]))
      const isAgent = await agentExists({ pubKey })
```

which throws an unexpected error on the JavaScript side that `Unrecognized object: [object Function]`.

I also tried without wrapping in `Buffer.from`, upon which the DHT tells me `Deserialize("invalid type: map, expected a HoloHash of primitive hash_type")`. Annoying! Not sure why this is a problem when  the same encoding is used for other composite ID fields.

No more time to investigate now, maybe we can do some pair programming on it in my AM if you're still up and about :) This might also be an opportune time to rip all my hacky custom logic out of `connection.ts` and replace with the more well-vetted stuff in https://github.com/holochain-open-dev/core-types/blob/main/src/utils.ts
